src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,117 @@ class JoinClauseSegment(BaseSegment):
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                Ref("SingleIdentifierGrammar", optional=True),
+                # Column alias in VALUES clause
+                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
+            ),
+            Ref("QuotedLiteralSegment"),
+        ),
+    )
+
+
+@spark3_dialect.segment()
+class DelimitedValues(BaseSegment):
+    """A ``VALUES`` clause can be a sequence either of scalar values or tuple values.
+
+    We make no attempt to ensure that all records have the same number of columns besides the distinction between
+    all scalar or all tuple, so for instance ``VALUES (1,2), (3,4,5)`` will parse but is not legal SQL.
+
+
+    """
+
+    type = "delimited_values"
+    match_grammar = OneOf(Delimited(Ref("ScalarValue")), Delimited(Ref("TupleValue")))
+
+
+@spark3_dialect.segment()
+class ScalarValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has a single column.
+
+    Ex: ``VALUES 1,2,3``
+    """
+
+    type = "scalar_value"
+    match_grammar = OneOf(
+        Ref("LiteralGrammar"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+    )
+
+
+@spark3_dialect.segment()
+class TupleValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has a multiple columns.
+
+    Ex: ``VALUES (1,2), (3,4)``
+    """
+
+    type = "tuple_value"
+    match_grammar = Bracketed(
+        Delimited(
+            Ref("ScalarValue"),
+        )
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class ValuesClauseSegment(BaseSegment):
+    """A ``VALUES`` clause, as typically used with ``INSERT`` or ``SELECT``.
+
+    The Spark SQL reference does not mention ``VALUES``` clauses except in the context of ``INSERT`` statements.
+    However, they appear to behave the same as in `postgres <https://www.postgresql.org/docs/13/sql-values.html>`.
+
+    In short, they can appear anywhere a ``SELECT`` can, and also as bare ``VALUES`` statements. Here are some examples:
+    .. code-block:: sql
+
+        VALUES 1,2 LIMIT 1;
+        SELECT * FROM VALUES (1,2) as t (a,b);
+        SELECT * FROM (VALUES (1,2) as t (a,b));
+        WITH a AS (VALUES 1,2) SELECT * FROM a;
+
+    """
+
+    type = "values_clause"
+    match_grammar = StartsWith(Ref.keyword("VALUES"))
+
+    parse_grammar = Sequence(
+        Ref.keyword("VALUES"),
+        DelimitedValues,
```

Comment:
Does this work? I thought you'd have to use `Ref("DelimitedValues")`.

Reply:
it appears to, isn't the `Ref` only necessary for something you've potentially redefined?

Reply:
Maybe. I have always used it for everything. @tunetheweb, @jpy-git: I remember talking about this recently. Did we know that direct class references work, at least sometimes?

Reply:
https://github.com/sqlfluff/sqlfluff/blob/43b0fc29f26194f8812a40ed6e55af57c8240da0/src/sqlfluff/core/parser/grammar/base.py#L85 seems to anticipate that `BaseGrammar`s or `BaseSegment`s might show up

type hints in the right places might prevent the hapless (myself for instance) from getting lucky if this is not intended to work

Reply:
The reason you need Ref is because the SQL grammar can often be recursive so using Ref allows the grammar to be lazily loaded therefore avoiding circular references

Reply:
So yes, best to just use it everywhere 👍

Reply:
@jpy-git in that case is there a reason not to throw (or fail type checking) if you get anything other than a `Ref` when instantiating a grammar?

like is there ever a case where using `Ref` is forbidden?